### PR TITLE
Few of improvement for class Counter construction

### PR DESCRIPTION
**Pull request by josephyu19850119**

### CODE REVIEW

File: igor_src/Counter.h [hunk 1]
```diff
@@ -73,7 +71,7 @@ class Counter {
 protected:
 	std::string path_to_file;
 	bool last_iter_only;
-	bool fstreams_created;
+	bool fstreams_created = false;
```
Comment (qmarcou) [inline igor_src/Counter.h]: I do not fully agree with this change. `fstreams_created = false` is the default initialization value, and only equals to false upon de novo creation of a Counter. Upon copy of an initialized counter this value is actually initialized to `true`. Thus I find default value assignment in the header misleading as it suggests that this is some kind of constant value. 
I think the following two discussions are interesting to justify my point of view: 
- https://stackoverflow.com/questions/28413154/c-initializing-variables-in-header-vs-with-constructor
- https://stackoverflow.com/questions/13662441/c11-allows-in-class-initialization-of-non-static-and-non-const-members-what-c

Comment (josephyu19850119) [reply igor_src/Counter.h]: I see, just keep your style.
Another similar issue in class Deletion constructor, there is lots of variable initialization. I'm intent to change them to In-class initialization, to significant improvement code readability. Anyway, I will share another PR, regardless you take it.

**Comment by josephyu19850119** [comment]
I have make another PR as your suggestion :)

**Comment by qmarcou** [comment]
Replaced by PR #56

**Pull request closed without merging**